test/modules/fundingManager/oracle/FM_PC_ExternalPrice_Redeeming_v1_Test.t.sol

```diff
@@ -0,0 +1,2012 @@
+
+import {
+    RedeemingBondingCurveBaseV1Mock,
+    IRedeemingBondingCurveBase_v1
+} from
+    "test/modules/fundingManager/bondingCurve/utils/mocks/RedeemingBondingCurveBaseV1Mock.sol";
+import {BancorFormula} from "@fm/bondingCurve/formulas/BancorFormula.sol";
+import {
+    PP_Streaming_v1,
+    IPP_Streaming_v1
+} from "src/modules/paymentProcessor/PP_Streaming_v1.sol";
+
+/**
+ * @title FM_PC_ExternalPrice_Redeeming_v1_Test
+ * @notice Test contract for FM_PC_ExternalPrice_Redeeming_v1
+ */
+contract FM_PC_ExternalPrice_Redeeming_v1_Test is ModuleTest {
+
+    // ═══════════════════════════════════════════════════════════════════════════════════════════════════════
+    // Storage
+    // ═══════════════════════════════════════════════════════════════════════════════════════════════════════
+
+    FM_PC_ExternalPrice_Redeeming_v1 fundingManager;
+    AuthorizerV1Mock authorizer;
+    RedeemingBondingCurveBaseV1Mock bondingCurveFundingManager;
+
+    // Test addresses
+    address admin;
+    address user;
+    address whitelisted;
+    address queueManager;
+
+    // Mock tokens
+    ERC20Issuance_v1 issuanceToken;    // The token to be issued
+
+    // Mock oracle
+    LM_ManualExternalPriceSetter_v1 oracle;
+
+    // Payment processor
+    PP_Streaming_v1AccessMock paymentProcessor;
+    ERC20PaymentClientBaseV1Mock paymentClient;
+
+
+    // Constants
+    string internal constant NAME = "Issuance Token";
+    string internal constant SYMBOL = "IST";
+    uint8 internal constant DECIMALS = 18;
+    uint internal constant MAX_SUPPLY = type(uint).max;
+    bytes32 constant WHITELIST_ROLE = "WHITELIST_ROLE";
+    bytes32 constant ORACLE_ROLE = "ORACLE_ROLE";
+    bytes32 constant QUEUE_MANAGER_ROLE = "QUEUE_MANAGER_ROLE";
+    uint8 constant INTERNAL_DECIMALS = 18;
+    uint constant BPS = 10000; // Basis points (100%)
+
+    // Fee settings
+    uint constant DEFAULT_BUY_FEE = 100;     // 1%
+    uint constant DEFAULT_SELL_FEE = 100;    // 1%
+    uint constant MAX_BUY_FEE = 500;         // 5%
+    uint constant MAX_SELL_FEE = 500;        // 5%
+    bool constant DIRECT_OPERATIONS_ONLY = false;
+
+    // Module Constants
+    uint constant MAJOR_VERSION = 1;
+    uint constant MINOR_VERSION = 0;
+    uint constant PATCH_VERSION = 0;
+    string constant URL = "https://github.com/organization/module";
+    string constant TITLE = "Module";
+
+    bytes32 internal roleId;
+    bytes32 internal roleIDOracle; 
+    bytes32 internal queueManagerRoleId;
+
+    uint private constant BUY_FEE = 0;
+    uint private constant SELL_FEE = 0;
+    bool private constant BUY_IS_OPEN = true;
+    bool private constant SELL_IS_OPEN = true;
+
+    address formula;
+
+
+    // ═══════════════════════════════════════════════════════════════════════════════════════════════════════
+    // Setup
+    // ═══════════════════════════════════════════════════════════════════════════════════════════════════════
+
+    function setUp() public {
```

Comment:
Most of the setup should be done differently and has been implemented in the PR I opened